### PR TITLE
Integration weekend tunes

### LINKED ISSUE #309: Dynup: First walkready pose is broken
When going into walkready after the power has been turned on, the IK solution for the walkready pose is bad, sometimes leading to the feet interlocking or joints colliding. This potentially causes damage to the robot. This seems more likely to happen when joints are outside of the accepted range, but this needs to be further evaluated.

## Expected behavior

Directly move to the walkready pose.

## Current behavior

Move to walkready through weird paths.

## Steps to Reproduce

1. Turn off motor power and stop all software
2. Turn on motor power
3. Start motion_standalone.launch
...

## Context (Environment)

- [ ] RViz
- [ ] Simulator
- [ ] Robot
- [ ] Local

### LINKED ISSUE #321: Dynup: Use joint space trajectories for arms

### LINKED ISSUE #496: Dynup: move feet closer to each other
The feet are closer together in walkready so we fall at the first step. Also we never fall to the side so we could do it without decreasing stability.

### LINKED ISSUE #520: Fix Walk ready pose
# Summary

## Steps to Reproduce

1.
2.
3.

### PR DESCRIPTION
**Pull request by Flova**
## Proposed changes

Closes #496 
Closes #321
Closes #309
Closes #520

## Checklist

- [x] Run `colcon build`
- [x] Write documentation
- [ ] Test on your machine
- [ ] Test on the robot
- [ ] Create issues for future work
- [x] Triage this PR and label it

### CODE REVIEW

**Comment by Flova** [comment]
I cleaned up the standup animation a bit, so I would prefer testing it again on the real hardware before merging.

**Comment by jaagut** [comment]
Does this also close #520 ?

File: bitbots_motion/bitbots_hcm/bitbots_hcm/hcm_dsd/actions/play_animation.py [hunk 1]
```diff
@@ -131,6 +131,18 @@ def choose_animation(self):
...
+class PlayAnimationStandupFront(AbstractPlayAnimation):
+    def choose_animation(self):
+        self.blackboard.node.get_logger().info("STANDUP FRONT ANIMATION")
+        return "stand_up_front"
+
+
+class PlayAnimationStandupBack(AbstractPlayAnimation):
+    def choose_animation(self):
+        self.blackboard.node.get_logger().info("STANDUP BACK ANIMATION")
+        return "stand_up_back"
```
Comment (jaagut) [inline bitbots_motion/bitbots_hcm/bitbots_hcm/hcm_dsd/actions/play_animation.py:143]: These return values are hardcoded. Please consider using the pattern used above.

File: bitbots_motion/bitbots_dynup/src/dynup_ik.cpp [hunk 2]
```diff
@@ -19,11 +19,23 @@ void DynupIK::init(moveit::core::RobotModelPtr kinematic_model) {
...
-  std::vector<std::string> names_vec = {"LHipPitch", "LKnee", "LAnklePitch", "RHipPitch", "RKnee", "RAnklePitch"};
-  std::vector<double> pos_vec = {0.7, 1.0, -0.4, -0.7, -1.0, 0.4};
-  for (size_t i = 0; i < names_vec.size(); i++) {
-    // besides its name, this method only changes a single joint position...
-    goal_state_->setJointPositions(names_vec[i], &pos_vec[i]);
+
+  // Use the current joint state as a starting point for the IK
+  if (joint_state_) {
+    for (size_t i = 0; i < joint_state_->name.size(); i++) {
+      // besides its name, this method only changes a single joint position...
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_motion/bitbots_dynup/src/dynup_ik.cpp:26]: the moveit interface is called joint **positions** but only sets a single position. The comment was also there before.

File: bitbots_motion/bitbots_hcm/bitbots_hcm/hcm_dsd/actions/play_animation.py (hunk 1 above)
Comment (Flova) [reply bitbots_motion/bitbots_hcm/bitbots_hcm/hcm_dsd/actions/play_animation.py:143]: Oh right

**Comment by Flova** [comment]
> Does this also close #520 ?

No not yet.

**Review by jaagut — APPROVED** [review]

**Pull request merged**